zerver/actions/streams.py

```diff
@@ -269,15 +270,17 @@ def do_unarchive_stream(stream: Stream, new_name: str, *, acting_user: UserProfi
 
     recent_traffic = get_streams_traffic({stream.id}, realm)
 
-    subscribed_users = {sub.user_profile for sub in stream_subscribers}
-    admin_users_and_bots = set(realm.get_admin_users_and_bots())
+    subscribed_user_ids = {sub.user_profile.id for sub in stream_subscribers}
+    admin_user_and_bot_ids = {user.id for user in realm.get_admin_users_and_bots()}
 
-    notify_users = admin_users_and_bots | subscribed_users
+    notify_user_ids = list(
+        admin_user_and_bot_ids
+        | subscribed_user_ids
+        | get_user_ids_with_metadata_access_via_permission_groups(stream)
+    )
```

Comment:
Why isn't this using can_access_stream_metadata_user_ids?

Reply:
can_access_stream_metadata_user_ids includes non-subscribers for public channels, while this union does not. I haven't looked into whether we want to notify non-subscribers or not and just preserved the original behaviour

Reply:
I think it makes sense to notify everyone with metadata access to the channel, since they should be able to see it in that case.